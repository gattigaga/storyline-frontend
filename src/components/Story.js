import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const Container = styled.div`
  width: 144px;
`;

const Thumbnail = styled.img`
  width: 144px;
  height: 192px;
  object-fit: cover;
  margin-bottom: 16px;
`;

const Title = styled.h1`
  font-family: Roboto;
  font-weight: bold;
  font-size: 18px;
  color: #333;
  margin-top: 0px;
  margin-bottom: 20px;
`;

const Author = styled.p`
  font-family: Roboto;
  font-size: 14px;
  color: #aaa;
  margin: 0px;
`;

const Date = styled.p`
  font-family: Roboto;
  font-size: 14px;
  color: #e74c3c;
  margin: 0px;
`;

const Story = ({ image, title, href, author, date }) => (
  <StyledLink to={href}>
    <Container>
      <Thumbnail src={image} />
      <Date>{date}</Date>
      <Title>{title}</Title>
      <Author>by {author}</Author>
    </Container>
  </StyledLink>
);

Story.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  href: PropTypes.string,
  author: PropTypes.string,
  date: PropTypes.string
};

Story.defaultProps = {
  href: "#"
};

export default Story;
