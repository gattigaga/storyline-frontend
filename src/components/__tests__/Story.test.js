import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Story from "../Story";

describe("Story", () => {
  const setup = propOverrides => {
    const props = {
      image: "dummy.png",
      title: "Introduction to React",
      author: "Gattigaga Hayyuta Dewa",
      date: "Jan 01, 2017",
      ...propOverrides
    };

    const wrapper = shallow(<Story {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
