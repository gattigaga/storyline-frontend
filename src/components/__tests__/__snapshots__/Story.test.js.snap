// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story should renders default 1`] = `
<Styled(Link)
  to="#"
>
  <styled.div>
    <styled.img
      src="dummy.png"
    />
    <styled.p>
      Jan 01, 2017
    </styled.p>
    <styled.h1>
      Introduction to React
    </styled.h1>
    <styled.p>
      by 
      Gattigaga Hayyuta Dewa
    </styled.p>
  </styled.div>
</Styled(Link)>
`;
